### DISCUSSION TITLE
Passing an object to the authorize function via credentials

### DISCUSSION THREAD

**Discussion by sheptang** [nextauthjs/next-auth#2656] [Help] [answered]
### Question 💬
I'm trying to pass some of the user field values from the database to the session variable, using Credentials provider. Since I'm using MongoDB, I do not want to hard-code the fields, as some fields might not even exist for other users. That's why I'm trying the `signIn` callback to pass these values using an object and take care of rest of the validation on the callback side.

But unfortunately, I see nothing more than a string that looks like: `[object Object]`.

### How to reproduce ☕️
Let's assume we have an object like:
```
var someObject = {
    username: "username",
    email: "name.surname@domain.tld",
    something: "else"
}
```
If you try using this `await signIn("credentials", {redirect: false, obj: someObject});`
It will end up passed as `[object Object]` to the `authorize()` function.

### Contributing 🙌🏽
I am willing to contribute as long as this is a bug or a missing feature that needs to be added.

**Comment by balazsorban44** [comment, marked as answer]
Check out the docs https://next-auth.js.org/providers/credentials

you can submit any form fields you want to, but the object has to be flat and defined in the credentials field in your provider config.

Everything sent this way will be passed to the authorize callback as the first argument:

https://github.com/nextauthjs/next-auth/blob/main/src/server/routes/callback.js#L334-L340

**Reply by sheptang** [reply]
So, I looked at the docs already, the only thing I was missing was flattening the object. When I do, I still have the same result.

The object I'm trying to pass looks like this (after flattening):
```js
{
    "0": "user",
    "_id": "613519c0cd1b7df6b992b10d",
    "email": "deneme@deneme.deneme",
    "username": "deneme"
}
```
I'm trying to pass this via:
```js
await signIn(
	"credentials",
	{
		redirect: false,
		user: flatUserObj
	}
);
```
Even the `redirect` property comes as a string `"false"`.
And my [...nextauth].js looks like this:
```js
export default NextAuth({
	providers: [
		Providers.Credentials({
			async authorize(credentials){
				const {user} = credentials;
				console.log(credentials);
				return user;
			}
		})
	],
	callbacks: {
		async signIn(user){
			if(user){
				// user variable comes from authorize output
				return user;
			}
		},
		async session(session){
			return session;
		},
		async jwt(token){
			return token;
		}
	}
});
```
I'm still having the same problem. Am I missing something? Maybe it's because I'm writing it all in JavaScript syntax, instead of TypeScript?

**Reply by balazsorban44** [reply]
Try
```js
await signIn("credentials", { redirect: false, ...flatUserObj })
```

I think redirect becoming a string is because sending params in `POST` will be parsed with `body-parser` internally, which might cause this. Just a guess though.

**Reply by sheptang** [reply]
This works perfect. Thank you!
